miden-lib/asm/sat/layout.masm

```diff
@@ -0,0 +1,396 @@
+
+# The memory address at which the latest known block hash is stored
+const.BLK_HASH_PTR=10
+
+# The memory address at which the account id is stored
+const.ACCT_ID_PTR=11
+
+# The memory address at which the initial account hash is stored
+const.INIT_ACCT_HASH_PTR=12
+
+# The memory address at which the global nullifier commitment is stored
+const.NULLIFIER_COM_PTR=13
+
+# The memory address at which the initial nonce is stored
+const.INIT_NONCE_PTR=14
+
+# ACCOUNT DATA
+# -------------------------------------------------------------------------------------------------
+
+# The memory address at which the account data section begins
+const.ACCT_DATA_SECTION_OFFSET=100
+
+# The memory address at which the account id and nonce is stored.
+# The account id is stored in the first element.
+# The account nonce is stored in the fourth element.
+const.ACCT_ID_AND_NONCE_PTR=100
+
+# The memory address at which the account vault root is stored
+const.ACCT_VAULT_ROOT_PTR=101
+
+# The memory address at which the account storage root is stored
+const.ACCT_STORAGE_ROOT_PTR=102
+
+# The memory address at which the account code root is stored
+const.ACCT_CODE_ROOT_PTR=103
+
+# CONSUMED NOTES DATA
+# -------------------------------------------------------------------------------------------------
+
+# The memory address at which the consumed note section begins
+const.CONSUMED_NOTE_SECTION_OFFSET=1000
+
+# The memory address at which the number of consumed notes is stored
+const.CONSUMED_NOTE_NUM_PTR=1000
+
+# The offsets at which data of a consumed note is stored relative to the start of its data segment
+const.CONSUMED_NOTE_HASH_OFFSET=0
+const.CONSUMED_NOTE_SERIAL_NUM_OFFSET=1
+const.CONSUMED_NOTE_SCRIPT_ROOT_OFFSET=2
+const.CONSUMED_NOTE_INPUTS_HASH_OFFSET=3
+const.CONSUMED_NOTE_VAULT_ROOT_OFFSET=4
+const.CONSUMED_NOTE_NUM_ASSETS_OFFSET=5
+const.CONSUMED_NOTE_ASSETS_OFFSET=6
+
+# CREATED NOTES
+# -------------------------------------------------------------------------------------------------
+
+# The memory address at which the created notes section begins.
+const.CREATED_NOTE_SECTION_OFFSET=10000
+
+# The offsets at which data of a created note is stored relative to the start of its data segment.
+const.CREATED_NOTE_HASH_OFFSET=0
+const.CREATED_NOTE_META_OFFSET=1
+const.CREATED_NOTE_RECIPIENT_OFFSET=2
+const.CREATED_NOTE_VAULT_HASH_OFFSET=3
+const.CREATED_NOTE_NUM_ASSETS_OFFSET=4
+const.CREATED_NOTE_ASSETS_OFFSET=5
+
+# MEMORY PROCEDURES
+# =================================================================================================
+
+# BOOK KEEPING
+# -------------------------------------------------------------------------------------------------
+
+#! Returns the number of created notes.
+#!
+#! Stack: []
+#! Output: [num_created_notes]
+export.get_num_created_notes
+    push.NUM_CREATED_NOTES_PTR mem_load
+end
+
+#! Returns a pointer to the current consumed note.
+#!
+#! Stack: []
+#! Output: [ptr]
+export.get_current_consumed_note
```

Comment:
The naming is a bit confusing - should we clarify (here and in other places) what "current consumed note" means? If I understand correctly, it is a note which is being executed, right?

Reply:
I think this is good. I wonder if in the future we should rename it to something even more clear - e.g., maybe something like `ACTIVE_NOTE`.